File: Generates_subsets.py
def subsets(s, curr, i):
    if (i == len(s)):
        print(curr)
        return
    subsets(s, curr, i+1)
    subsets(s, curr+s[i], i+1)

def main():
    s = str(input())
    curr = ""
    i = 0

    subsets(s, curr, i+1)
    subsets(s, curr+s[i], i+1)
main()

File: Generates_subsets.java
import java.util.Scanner;

public class Generates_subsets 
{
    public static void subsets(String s, String curr, int i)
    {
        if (i == s.length())
        {
            System.out.println(curr);
            return;
        }
        subsets(s, curr, i+1);
        subsets(s, curr + s.charAt(i), i+1);
    }
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine();

        String curr = "";
        int i = 0;
        subsets(s, curr, i);
        subsets(s, curr + s.charAt(i), i+1);
        // System.out.println(res);
    }
}
